Movecraft/src/main/java/net/countercraft/movecraft/commands/MovecraftCommandManager.java

```diff
@@ -0,0 +1,61 @@
+package net.countercraft.movecraft.commands;
+
+import co.aikar.commands.CommandIssuer;
+import co.aikar.commands.PaperCommandManager;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.regex.Pattern;
+
+public class MovecraftCommandManager extends PaperCommandManager {
+    public MovecraftCommandManager(Plugin plugin) {
+        super(plugin);
+    }
+
+    private static final Pattern COMMA = Pattern.compile(",");
+    private static final Pattern PIPE = Pattern.compile("\\|");
+
+    @Override
+    public boolean hasPermission(CommandIssuer issuer, String permission) {
+        if (permission == null || permission.isEmpty()) {
+            return true;
+        }
+
+        //handle AND like normal using comma ","
+        String[] perms = COMMA.split(permission);
+        if (perms.length > 1) {
+            return super.hasPermission(issuer, new HashSet<>(Arrays.asList(perms)));
+        }
+
+        //handle OR using pipe "|"
+        CommandSender sender = issuer.getIssuer();
+        for (String perm : PIPE.split(permission)) {
+            perm = perm.trim();
+            if (!perm.isEmpty() && senderHasPermission(sender, perm)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```

Comment:
I feel like there should be some sort of default handler for something like this.

Reply:
No went on acf discord and it isn't supported, one of the main dev said he used this approach in one of his plugins, also should be PRd to original acf but they are busy on other projects it looks like

Reply:
I still do not understand why we are overriding this function.  `CommandManager` has it's own implementation that should be sufficient, we do not need to add special logic to permissions with commas and pipes.

Reply:
Some commands checks 2 permissions if you see the way old commands were checked, the comma functionality was supported by the original PaperCommandManager, so nothing new there just had to reimplement it when i overrode the method, the problem was that some commands check for movecraft.command or movecraft.command.specificCommand for example, the AND was supported with commas, the OR not so had to add the feature

Reply:
None of our previous commands checked for one permission or another, they all required either a single permission or both `movecraft.commands` and their specific permission.